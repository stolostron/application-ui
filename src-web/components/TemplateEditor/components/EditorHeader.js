/*******************************************************************************
 * Licensed Materials - Property of IBM
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Tab, Tabs, Checkbox } from 'carbon-components-react'
import '../scss/editor-header.scss'
import msgs from '../../../../nls/platform.properties'

class EditorHeader extends React.Component {
  static propTypes = {
    children: PropTypes.node,
    handleShowSecretChange: PropTypes.func,
    handleTabChange: PropTypes.func,
    locale: PropTypes.string,
    otherYAMLTabs: PropTypes.array,
    showSecrets: PropTypes.bool,
    type: PropTypes.string.isRequired
  };

  constructor(props) {
    super(props)
  }

  render() {
    const { children, otherYAMLTabs = [], locale } = this.props
    const editorToolbarTitle = msgs.get('editor.toolbar', locale)
    const hasTabs = otherYAMLTabs.length > 0
    const classnames = classNames({
      'creation-view-yaml-header': true,
      hasTabs: hasTabs
    })
    return (
      <div className={classnames}>
        <div
          className="creation-view-yaml-header-toolbar"
          role="region"
          aria-label={editorToolbarTitle}
          id={editorToolbarTitle}
        >
          {children}
        </div>
        {hasTabs && (
          <div className="creation-view-yaml-header-tabs">
            {this.renderEditorTabs(otherYAMLTabs)}
            {this.renderShowSecrets()}
          </div>
        )}
      </div>
    )
  }

  renderEditorTabs = otherYAMLTabs => {
    const { type = 'unknown', handleTabChange } = this.props
    return (
      <Tabs selected={0} aria-label={'Select template'}>
        <Tab
          label={type}
          key="main"
          id="main"
          onClick={handleTabChange.bind(this, 0)}
        />
        {otherYAMLTabs.map(({ id }, inx) => {
          return (
            <Tab
              label={id}
              key={id}
              id={id}
              onClick={handleTabChange.bind(this, inx + 1)}
            />
          )
        })}
      </Tabs>
    )
  };

  renderShowSecrets = () => {
    const { showSecrets, handleShowSecretChange, locale } = this.props
    return (
      <div className="creation-view-yaml-header-secrets">
        <Checkbox
          id="show-secrets"
          className="checkbox"
          hideLabel
          labelText=""
          checked={showSecrets}
          onChange={handleShowSecretChange}
        />
        <div>{msgs.get('editor.show.secrets', locale)}</div>
      </div>
    )
  };
}

export default EditorHeader
