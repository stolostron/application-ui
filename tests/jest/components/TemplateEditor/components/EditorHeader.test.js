/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc.
*/
"use strict";

import React from "react";
import EditorHeader from "../../../../../src-web/components/TemplateEditor/components/EditorHeader";
import renderer from "react-test-renderer";

const type = "application";
const otherYAMLTabs = [];
const handleTabChange = jest.fn;

describe("EditorHeader component", () => {
  it("renders as expected", () => {
    const fn = jest.fn();
    const exceptions = [{ text: "bad", row: 0 }];
    const component = renderer.create(
      <EditorHeader
        otherYAMLTabs={otherYAMLTabs}
        handleTabChange={handleTabChange}
        type={type}
        locale={"en-US"}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
