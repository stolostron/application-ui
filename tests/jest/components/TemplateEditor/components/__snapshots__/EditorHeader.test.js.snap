// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorHeader component renders as expected 1`] = `
<div
  className="creation-view-yaml-header"
>
  <div
    className="creation-view-yaml-header-toolbar"
    role="region"
  />
</div>
`;
